src/main/java/org/generationcp/middleware/dao/GermplasmStudySourceDAO.java

```diff
@@ -0,0 +1,195 @@
+package org.generationcp.middleware.dao;
+
+import liquibase.util.StringUtils;
+import org.generationcp.middleware.pojos.GermplasmStudySource;
+import org.generationcp.middleware.pojos.SortedPageRequest;
+import org.generationcp.middleware.service.api.study.germplasm.source.StudyGermplasmSourceDto;
+import org.generationcp.middleware.service.api.study.germplasm.source.StudyGermplasmSourceRequest;
+import org.generationcp.middleware.util.SqlQueryParamBuilder;
+import org.hibernate.Criteria;
+import org.hibernate.SQLQuery;
+import org.hibernate.criterion.Projections;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.transform.Transformers;
+import org.hibernate.type.IntegerType;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class GermplasmStudySourceDAO extends GenericDAO<GermplasmStudySource, Integer> {
+
+	protected static final String GERMPLASM_STUDY_SOURCE_SEARCH_QUERY = "SELECT \n"
+		+ "gss.source_id as `sourceId`,\n"
+		+ "gss.gid as `gid`,\n"
+		+ "g.mgid as `groupId`,\n"
+		+ "n.nval as `designation`,\n"
+		+ "m.mcode as `breedingMethodAbbreviation`,\n"
+		+ "m.mname as `breedingMethodName`,\n"
+		+ "m.mtype as `breedingMethodType`,\n"
+		+ "geo.description as `trialInstance`,\n"
+		+ "loc.lname as `location`,\n"
+		+ "rep_no.value as `replicationNumber`,\n"
+		+ "plot_no.value as `plotNumber`,\n"
+		+ "g.gdate as `germplasmDate`,\n"
+		+ "count(lot.lotid) as `lots` "
+		+ "FROM germplasm_study_source gss \n"
+		+ "INNER JOIN germplsm g ON g.gid = gss.gid\n"
+		+ "INNER JOIN project p ON p.project_id = gss.project_id\n"
+		+ "LEFT JOIN nd_experiment e ON e.nd_experiment_id = gss.nd_experiment_id\n"
+		+ "LEFT JOIN nd_experimentprop rep_no ON rep_no.nd_experiment_id = e.nd_experiment_id AND rep_no.type_id = 8210\n"
```

Comment:
Use TermId.REP_NO, TermId.PLOT_NO and TermId.LOCATION_ID instead of the cvterm IDs directly in the query

Reply:
Done